src/gordon_gcp/clients/auth.py

```diff
@@ -101,6 +104,13 @@ def __init__(self, keyfile=None, scopes=None, session=None, loop=None):
         self.token = None
         self.expiry = None  # UTC time
 
+        if isinstance(self.creds, compute_engine.credentials.Credentials):
+            self.refresh_token_f = \
+                self._refresh_token_using_compute_credentials
+        else:
+            self.refresh_token_f = \
+                self._refresh_token_using_service_account_credentials
```

Comment:
I realize it may have been misleading in my previous comment when I said in "init", since it is actually a function called from init, but what I was trying to ask is if it would it be possible to set this function in `_load_credentials` to avoid the use of `isinstance`, since we already know what type of credentials we have at that point.

For example, just above the following lines:
- https://github.com/spotify/gordon-gcp/pull/118/files#diff-1f1b3f74e2435d498631282277981099L128
- https://github.com/spotify/gordon-gcp/pull/118/files#diff-1f1b3f74e2435d498631282277981099L133

Reply:
Ah ok, I understand. When not having a service account file set then the default credentials code is used:
```python
        credentials, _ = gauth.default(
            scopes=['https://www.googleapis.com/auth/userinfo.email'])
```
The problem is that depending on certain conditions (gcloud available, running on compute engine or not, etc) the credentials object returned will be different. For example, when running locally it would use gcloud to return the normal service account credentials with client id, client secret, refresh token, token uri and then we want to use the token uri (normal oauth refresh token call). 
When it returns the compute engine credentials because the service runs on GKE/GCE then we want to call the metadata service. So, the logic of which credentials object we get is hidden in the `gauth.default` call and it seems the only real way to know which one it is, is checking the type.

Reply:
Aha, I didn't realize there could be two relevant environments in that section. Thanks for the detailed info! I was a little confused when this would be used since all plugins currently require keyfiles [1] [2], but I guess local gcloud auth could be handy for local debugging once we start to move them off that requirement, so it makes sense to support it.

[1] https://gordon-dns-gcp.readthedocs.io/en/latest/config-service.html#cmdoption-arg-keyfile
[2] https://gordon-dns-gcp.readthedocs.io/en/latest/config-janitor.html#cmdoption-arg-keyfile